src/App.js

```diff
@@ -1,20 +1,82 @@
 import React from "react";
 import logo from "./logo.png";
 import "./App.css";
+import { TextField, Button, Grid, Typography } from "@mui/material";
+import axios from "axios";
 
-class App extends React.Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-        </header>
-      </div>
-    );
-  }
-}
+export default function App() {
+  const [city, setCity] = React.useState("");
+  const [weatherData, setWeatherData] = React.useState({
+    temperature: "",
+    weatherIcon: "",
+  });
+
+  const handleChange = (event) => {
+    setCity(event.target.value);
+  };
 
-export default App;
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    axios
+      .get(
+        `https://api.openweathermap.org/geo/1.0/direct?q=${city}&limit=1&appid=${process.env.React_App_weather_api}`
+      )
+      .then((response) => response.data[0])
+      .then((cityGeoData) =>
+        axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?lat=${cityGeoData.lat}&lon=${cityGeoData.lon}&appid=${process.env.React_App_weather_api}&units=metric`
+        )
+      )
+      .then((response) => {
+        const { data: weatherData } = response;
+        console.log(weatherData);
+        setWeatherData((prevWeatherData) => {
+          return {
+            ...prevWeatherData,
+            temperature: weatherData.main.temp,
+            weatherIcon: weatherData.weather[0].icon,
+          };
+        });
+      });
+  };
+
+  return (
+    <Grid container>
+      <Grid item xs={12}>
+        <img src={logo} className="App-logo" alt="logo" />
+      </Grid>
+      <Grid item xs={6} sx={{ mt: 2 }}>
+        <form sx={{ mt: 2 }} noValidate autoComplete="off">
+          <TextField
+            label="City"
+            variant="outlined"
+            onChange={handleChange}
+            value={city}
+          />
+        </form>
+      </Grid>
+      <Button
+        sx={{ color: "black", border: "black" }}
+        onClick={handleSubmit}
+        variant="outlined"
+      >
+        Check Weather
+      </Button>
+      {weatherData.temperature && (
+        <Grid item xs={6} sx={{ mt: 2 }}>
+          <img
+            src={
+              "https://openweathermap.org/img/w/" +
+              weatherData.weatherIcon +
+              ".png"
+            }
+            alt="weather pic"
+          />
+          <Typography variant="h6">
+            {weatherData.temperature} &#8451;
```

Comment:
Would be great if we could give `&#8451` a name via variable assignment. Like this very hard to tell what kind of icon it is.